### DISCUSSION TITLE
How to filter based on array that includes any of the values not all of them

### DISCUSSION THREAD

**Discussion by agcty** [prisma/prisma#18135] [Q&A]
Is it possible to write a query that returns any of the array_contains matches, and not all of them?

E.g I'd like to find all cars that have a color of either gray or white, not gray AND white.

```jsx
const getCars = await prisma.cars.findMany({
  where: {
    color: {
      array_contains: ['gray', "white"],
    },
  },
})
```

**Comment by nurul3101** [comment]
Hello @agcty 👋 

Considering this model:
```prisma
model Cars {
  id     Int      @id @default(autoincrement())
  name   String
  colors String[]
}
```

I was able to get the cars which contains either of the value through [hasSome](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#hassome) filter:
```typescript
import { PrismaClient, Prisma } from "@prisma/client";

const prisma = new PrismaClient();


// A `main` function so that you can use async/await
async function main() {
  await prisma.cars.create({
    data: {
      name: "BMW",
      colors: ["red", "blue"],
    },
  });

  await prisma.cars.create({
    data: {
      name: "Aston Martin",
      colors: ["yellow", "green"],
    },
  });

  const cars = await prisma.cars.findMany({
    where: {
      colors: {
        hasSome: ["red", "green"],
      },
    },
  });

  console.log(cars);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

Here's the response:

<img width="505" alt="Screenshot 2023-02-28 at 4 13 01 PM" src="https://user-images.githubusercontent.com/13049676/221895589-4349a39f-de23-4a7b-bf0c-9a2a141ea3ee.png">

**Comment by agcty** [comment]
Thanks @nurul3101, maybe I should have clarified that I'm using MySQL which doesn't have scalar attributes as far as I'm aware of :(

**Comment by dev-josias** [comment]
Hello. For anyone encountering the same issue. I found a workaround using Json field.
Here is what worked for me
```js
const colors = ['gray', 'white'];
const getCars = await prisma.cars.findMany({
  where: {
    OR: colors.map(color => (
         { colors: { array_contains: [color] } }
    ))
  },
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
